lib/annotations2triannon/open_annotation.rb

```diff
@@ -18,16 +18,19 @@ class OpenAnnotation
     # @param id [UUID|URI|String] to identify an open annotation
     def initialize(graph=RDF::Graph.new, id=nil)
       @@agent ||= Annotations2triannon::AGENT
-      raise ArgumentError, 'graph must be RDF::Graph instance' unless graph.instance_of? RDF::Graph
-      @graph = graph
-      @id = id.nil? ? get_id : RDF::URI.parse(id)
-      insert_annotation unless is_annotation?
+      set_graph(graph)
+      set_id(id) # must be set after setting the graph
+      insert_annotation unless is_annotation? # useful when graph.empty?
+    end
+
+    # @see #set_id
+    def id=(id)
```

Comment:
no, it's calling a method in the class with the passed param value;  that's allowed.  id=  is a method.   But it's okay if you don't want to.